indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/ParallelIndexSupervisorTask.java

```diff
@@ -169,6 +173,11 @@
   @MonotonicNonNull
   private volatile TaskToolbox toolbox;
 
+  private IndexTask sequentialIndexTask;
+  private ParallelIndexTaskRunner<SinglePhaseSubTask, PushedSegmentsReport> parallelSinglePhaseRunner;
```

Comment:
I suppose you can get the current runner or task from `currentSubTaskHolder` and cast it properly depending on `isParallelMode()` instead of adding these?

Reply:
I got rid of the references and adjusted things to use `currentSubTaskHolder`